src/translator_functions.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+###
+# Inspired by https://github.com/oscarotero/Gettext/blob/master/src/translator_functions.php
+###
+
+/**
+ * Noop, marks the string for translation but returns it unchanged.
+ */
+if (!function_exists('noop')) {
+    function noop($original)
+    {
+        return $original;
+    }
+}
+
+if (!function_exists('noop_')) {
+    function noop_($original)
+    {
+        return $original;
+    }
+}
+
+if (!function_exists('noop__')) {
+    function noop__($original)
+    {
+        return $original;
+    }
+}
+
+/**
+ * Returns the translation of a string without interpolation.
+ */
+if (!function_exists('_')) {
+    function _($original)
+    {
+        return gettext($original);
+    }
+}
+
+/**
+ * Returns the translation of a string with interpolation.
+ */
+if (!function_exists('i_')) {
+    function i_($original)
```

Comment:
Maybe its worth proxying these function to some class? Especially given that few of them have the same body.

Reply:
I tried but actually only 2 of them have the same body (`i_` and `i__`) and since there is strange things happening using `func_get_args()`, it's already more complex than just doing something like: 

```php
function i__($original)
{
  return i_($original);
}
```

because it will lose the extra (hidden) arguments.

I have to admit I don't care a lot about this specific duplication of code. That code will not evolve much and it's just 10 lines of code.